var/spack/repos/builtin/packages/duckdb/package.py

```diff
@@ -16,8 +16,106 @@ class Duckdb(CMakePackage):
     version("0.8.1", sha256="a0674f7e320dc7ebcf51990d7fc1c0e7f7b2c335c08f5953702b5285e6c30694")
+    version("0.8.0", sha256="df3b8e0b72bce38914f0fb1cd02235d8b616df9209beb14beb06bfbcaaf2e97f")
     version("0.7.1", sha256="67f840f861e5ffbe137d65a8543642d016f900b89dd035492d562ad11acf0e1e")
+
+    depends_on("python@3:")
+    depends_on("cmake")
+    depends_on("gmake")
+    depends_on("ninja", when="+ninjabuild")
+    depends_on("openssl", when="+httpfs")
+    depends_on("icu4c", when="~icu")
+
+    # Build Options
+    variant("autocomplete", default=True, description="Include autocomplete for CLI in build")
+    variant("cli", default=True, description="Compile with command line client")
+    variant("icu", default=False, description="Compile with bundled ICU library")
+    variant("ninjabuild", default=True, description="Use GEN=ninja to build")
+    variant("openssl", default=False, description="Compile with bundled OpenSSL library")
+
+    # Extensions
+    variant("excel", default=True, description="Include Excel formatting extension in build")
+    variant("fts", default=True, description="Include FTS (full text search) support in build")
+    variant("httpfs", default=True, description="Include HTTPFS (& S3) support in build")
+    variant("inet", default=True, description="Include INET (ip address) support in build")
+    variant("json", default=True, description="Include JSON support in build")
+    variant("parquet", default=True, description="Include parquent support in build")
+
+    # APIs
+    variant("jdbc", default=False, description="Build JDBC driver (may not work)")
+    variant("odbc", default=False, description="Build with ODBC driver (may not work)")
+    variant("python", default=False, description="Build with Python driver (may not work)")
+
+    def edit(self, spec, prefix):
+        if "+ninjabuild" in spec:
+            env["GEN"] = "ninja"
+        variant_flags = [
+            "autocomplete",
+            "cli",
+            "excel",
+            "fts",
+            "httpfs",
+            "icu",
+            "inet",
+            "jdbc",
+            "json",
+            "odbc",
+            "openssl",
+            "parquet",
+            "python",
+        ]
+        for flag in variant_flags:
+            make_flag = "BUILD_" + flag.upper()
+            if "+" + flag in spec:
+                env[make_flag] = "1"
+            elif "~" + flag in spec:
+                env[make_flag] = "0"
+
+    def url_for_version(self, version):
+        return "https://github.com/duckdb/duckdb/archive/refs/tags/v{0}.tar.gz".format(version)
+
+    def patch(self):
+        # DuckDB pulls its version from a git tag, which it can't find in the tarball
+        # and thus defaults to something arbitrary and breaks extensions.
+        # We use the Spack version to inject it in the right place during the build
+
+        v = self.spec.version
+        if v < Version("0.10.0"):
```

Comment:
The:
```python
with when(...):
    ...
```
block works only with directives. Within a method like this it's more idiomatic to:
```suggestion
        if not self.spec.satisfies("@0.10.0:"):
```